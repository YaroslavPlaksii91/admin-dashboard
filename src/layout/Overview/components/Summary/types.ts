export type SummaryItemType = {
  label: string;
  count: number;
};

export type SummaryProps = {
  data: SummaryItemType[];
};
