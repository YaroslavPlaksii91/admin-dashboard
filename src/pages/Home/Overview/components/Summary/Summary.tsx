import { FC } from 'react';
import { List, ListItem, Typography } from '@mui/material';

import { SummaryProps } from './types';

export const Summary: FC<SummaryProps> = ({ data }) => {
  return (
    <List sx={{ display: 'flex', gap: '30px', mb: '30px' }}>
      {data.map(item => (
        <ListItem
          key={item.label}
          sx={{
            display: 'flex',
            flexDirection: 'column',
            padding: '24px 32px',
            border: '1px solid',
            borderColor: 'borderColor',
            borderRadius: '8px',
            color: 'secondaryTextColor',
            backgroundColor: 'mainWhiteColor',
            transition: 'color 250ms ease, border-color 250ms ease',
            '&:hover': {
              color: 'accentBlueColor',
              borderColor: 'accentBlueColor',
              cursor: 'pointer',
              span: {
                color: 'accentBlueColor',
              },
            },
          }}
        >
          <Typography variant="h3" sx={{ mb: '12px' }}>
            {item.label}
          </Typography>
          <Typography
            component="span"
            sx={{
              fontSize: '40px',
              fontWeight: 700,
              letterSpacing: '1px',
              color: 'mainTextColor',
              transition: 'color 250ms ease',
            }}
          >
            {item.count}
          </Typography>
        </ListItem>
      ))}
    </List>
  );
};
