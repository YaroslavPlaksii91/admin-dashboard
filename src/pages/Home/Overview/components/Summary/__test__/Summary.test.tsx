import { cleanup, render, screen } from '@testing-library/react';

import { Summary } from '../Summary';

describe('Summary component', () => {
  afterEach(cleanup);

  const mockData = [
    { label: 'Label 1', count: 10 },
    { label: 'Label 2', count: 20 },
    { label: 'Label 3', count: 30 },
  ];

  it('should be rendered', () => {
    const { asFragment } = render(<Summary data={mockData} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders each item in the data', () => {
    render(<Summary data={mockData} />);
    const listItems = screen.getAllByRole('listitem');

    expect(listItems).toHaveLength(mockData.length);
  });

  it('displays label and count for each item', () => {
    render(<Summary data={mockData} />);

    mockData.forEach(item => {
      expect(screen.getByText(item.label)).toBeInTheDocument();
      expect(screen.getByText(item.count.toString())).toBeInTheDocument();
    });
  });
});
