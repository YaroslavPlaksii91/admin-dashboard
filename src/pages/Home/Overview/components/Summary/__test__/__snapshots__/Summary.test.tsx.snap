// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary component should be rendered 1`] = `
<DocumentFragment>
  <ul
    class="MuiList-root MuiList-padding css-1ixztv0-MuiList-root"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1vni9sx-MuiListItem-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 css-wy4g72-MuiTypography-root"
      >
        Label 1
      </h3>
      <span
        class="MuiTypography-root MuiTypography-body1 css-zun636-MuiTypography-root"
      >
        10
      </span>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1vni9sx-MuiListItem-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 css-wy4g72-MuiTypography-root"
      >
        Label 2
      </h3>
      <span
        class="MuiTypography-root MuiTypography-body1 css-zun636-MuiTypography-root"
      >
        20
      </span>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1vni9sx-MuiListItem-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 css-wy4g72-MuiTypography-root"
      >
        Label 3
      </h3>
      <span
        class="MuiTypography-root MuiTypography-body1 css-zun636-MuiTypography-root"
      >
        30
      </span>
    </li>
  </ul>
</DocumentFragment>
`;
